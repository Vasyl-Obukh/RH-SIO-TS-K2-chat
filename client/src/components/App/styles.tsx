import styled from 'styled-components';

export const Container = styled.div`
  display: grid;
  min-height: 100vh;
  grid-template-rows: auto 1fr auto;
  overflow-x: hidden;
  font-family: ${props => props.theme && props.theme.fontFamily ? props.theme.fontFamily : 'Arial, sans-serif'};
`;