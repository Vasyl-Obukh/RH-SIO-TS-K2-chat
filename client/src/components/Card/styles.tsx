import styled from 'styled-components';
import { Link } from 'react-router-dom';

export const Wrapper = styled.li`
  display: flex;
  align-items: center;
  justify-content: center;
  margin: 15px 10px;
  height: 125px;
  width: 250px;
  list-style: none;
  border-radius: 25px;
  border: 4px solid ${props => props.theme.primaryColor || '#000000'};
  background-color: ${props => props.theme.primaryReverseColor || '#ffffff'};
  color: ${props => props.theme.primaryColor || '#000000'};
  transition: all .2s;
  cursor: pointer;
  
  &:hover {
    transform: scale(1.1);
  }
`;

export const Title = styled(Link)`
  text-decoration: none;
  color: inherit;
  font-size: 24px;
  font-weight: bold;
`;