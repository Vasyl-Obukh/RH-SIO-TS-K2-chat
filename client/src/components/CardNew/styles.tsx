import styled from 'styled-components';

export const Container = styled.div`
  padding: 15px 0;
`;

export const Form = styled.form`
  position: relative;
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 40px 0 25px;
  width: 300px;
  border-radius: 25px;
  border: 4px solid ${props => props.theme.primaryColor || '#000000'};
  color: ${props => props.theme.primaryReverseColor || '#ffffff'};
`;

export const Input = styled.input<any>`
  box-sizing: border-box;
  margin: 15px 0;
  padding: 10px 15px;
  width: 250px;
  border: 2px solid ${props => props.theme.primaryColor || '#000000'};
  outline: none;
  border-radius: 25px;
`;

export const Button = styled.button<any>`
  margin: 0 auto;
  padding: 10px 0;
  width: 250px;
  border: none;
  outline: none;
  border-radius: 25px;
  background-color: ${props => props.theme.primaryColor || '#000000'};
  color: ${props => props.theme.primaryReverseColor || '#ffffff'};
  text-align: center;
  transition: all .2s;
  cursor: pointer;
  
  &:hover {
    transform: scale(1.05);
    opacity: .85;
  }
`;

export const Close = styled.span`
  position: absolute;
  top: 0;
  right: 10px;
  font-size: 50px;
  line-height: 45px;
  color: ${props => props.theme.primaryColor || '#000000'};
  cursor: pointer;
  transition: all .2s;
  
  &:hover {
    transform: scale(1.2);
    opacity: .75;
  }
`;