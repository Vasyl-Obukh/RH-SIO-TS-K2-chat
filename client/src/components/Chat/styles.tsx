import styled from 'styled-components';

export const Container = styled.div`
  border-radius: 25px;
  border: 2px solid ${props => props.theme.primaryColor || '#000000'}
`;