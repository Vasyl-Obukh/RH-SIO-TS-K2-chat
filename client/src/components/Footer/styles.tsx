import styled from 'styled-components';

export const Container = styled.div`
  padding: 10px 50px;
  background-color: ${props => props.theme && props.theme.primaryColor ? props.theme.primaryColor : '#000000'};
  color: ${props => props.theme && props.theme.primaryReverseColor ? props.theme.primaryReverseColor : '#ffffff'};
`;

export const Copyright = styled.p`
  text-align: center;
  color: inherit;
  font-style: italic;
  font-size: 14px;
`;