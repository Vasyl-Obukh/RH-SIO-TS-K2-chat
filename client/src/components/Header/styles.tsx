import styled from 'styled-components';
import { Link as RouterLink } from 'react-router-dom';

export const Container = styled.div`
  display: flex;
  justify-content: space-between;
  padding: 25px 50px;
  background-color: ${props => props.theme && props.theme.primaryColor ? props.theme.primaryColor : '#000000'};
  color: ${props => props.theme && props.theme.primaryReverseColor ? props.theme.primaryReverseColor : '#ffffff'};
`;

export const Link = styled(RouterLink)<any>`
  color: inherit;
  font-size: 24px;
  font-weight: ${props => props.as === 'span' ? 'normal' : 'bold'};
  font-style: ${props => props.as === 'span' ? 'italic' : 'initial'};
  text-decoration: none;
  transition: all .2s;
  cursor: pointer;
  
  &:hover {
    text-decoration: underline;
  }
`;