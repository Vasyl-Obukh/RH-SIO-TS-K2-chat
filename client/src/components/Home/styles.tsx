import styled from 'styled-components';

export const Container = styled.div`
  display: grid;
  grid-template-rows: auto 1fr;
  min-height: 100%;
  padding: 15px 35px;
`;