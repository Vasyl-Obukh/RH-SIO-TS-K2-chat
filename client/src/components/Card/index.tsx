import * as React from 'react';
import * as Styles from './styles';

export interface CardInterface {
  title: string,
  url: string
}

function Card(props: CardInterface) {
  return (
    <Styles.Wrapper>
      <Styles.Title to={props.url}>{props.title}</Styles.Title>
    </Styles.Wrapper>
  );
}

export default Card;