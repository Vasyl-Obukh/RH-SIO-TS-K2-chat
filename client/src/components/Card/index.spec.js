import React from 'react';
import { shallow } from 'enzyme';
import Card from './index';
import * as Styles from './styles';

describe('<Card /> tests', () => {
  const props = {
    title: 'test title',
    url: 'test url'
  };

  test('Should render correctly', () => {
    const component = shallow(<Card {...props} />);
    expect(component).toMatchSnapshot();
  });

  test('Should pass url to the Link\'s attribute "to"', () => {
    const component = shallow(<Card {...props} />);
    expect(component.find(Styles.Title).prop('to')).toBe('test url');
  })

  test('Should pass title as the Link\'s child', () => {
    const component = shallow(<Card {...props} />);
    expect(component.find(Styles.Title).text()).toBe('test title');
  })
});