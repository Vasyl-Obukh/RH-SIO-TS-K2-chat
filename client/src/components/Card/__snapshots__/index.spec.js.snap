// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> tests Should render correctly 1`] = `
<styled.li>
  <Styled(Link)
    to="test url"
  >
    test title
  </Styled(Link)>
</styled.li>
`;
